src/rows.ts

```diff
@@ -39,62 +38,62 @@ export class Rows {
   }
 
   /**
-   * Returns a readable stream of {@link Row}s for responses
+   * Returns an async iterator of {@link Row}s for responses
    * in {@link StreamableDataFormat} format.
    *
-   * The method will throw if called on a response in non-streamable format,
-   * and if the underlying stream was already consumed
-   * by calling the other methods
+   * If selected format is non-streamable,
+   * or the underlying stream was already consumed,
+   * it will throw when accessing the next element
    */
-  stream(): Stream.Readable {
-    // If the underlying stream has already ended by calling `text` or `json`,
-    // Stream.pipeline will create a new empty stream
-    // but without "readableEnded" flag set to true
+  async *stream(): AsyncGenerator<Row, void> {
     if (this._stream.readableEnded) {
       throw Error(streamAlreadyConsumedMessage)
     }
-
     validateStreamFormat(this.format)
+    const textDecoder = new TextDecoder()
+    let decodedChunk = ''
+    for await (const chunk of this._stream) {
+      decodedChunk += textDecoder.decode(chunk, { stream: true })
+      let idx = 0
+      while (true) {
+        idx = decodedChunk.indexOf('\n')
+        if (idx !== -1) {
+          const line = decodedChunk.slice(0, idx)
+          decodedChunk = decodedChunk.slice(idx + 1)
+          yield {
+            /**
+             * Returns a string representation of a row.
+             */
+            text(): string {
```

Comment:
Are you sure it's more lightweight than a `Row` class instance?
The method emits an object literal with 2 methods vs. a class with 2 methods in a prototype
```js
class Row {
    text(){}
}
(new Row()).hasOwnProperty('text') // false
const objectLiteral = { 
    text(){}
}
objectLiteral.hasOwnProperty('text') // true
```

Reply:
```ts
split((row: string) => new Row(row, 'JSON')),
```
```bash
ts-node --transpile-only --project tsconfig.dev.json examples/many_numbers.ts
Execution time: 1957 ms
```

```ts
split((row: string) => ({
  text: row,
  json<T>() {
    return decode(row, 'JSON')
  }
})),
```

```bash
ts-node --transpile-only --project tsconfig.dev.json examples/many_numbers.ts
Execution time: 394 ms
```